Reporting/ad-extensions-reporter.js

```diff
@@ -0,0 +1,395 @@
+                if (disapprovalShortNamesByExtensionType[placeholderTypeId][
+                    jsonAttributeValuesByExtensionType[placeholderTypeId].indexOf(jsonAttributeValues)
+                ].indexOf(disapprovalName) === -1) {
+                    disapprovalShortNamesByExtensionType[placeholderTypeId][
+                        jsonAttributeValuesByExtensionType[placeholderTypeId].indexOf(jsonAttributeValues)
+                    ] = disapprovalShortNamesByExtensionType[placeholderTypeId][
+                        jsonAttributeValuesByExtensionType[placeholderTypeId].indexOf(jsonAttributeValues)
+                    ].concat(disapprovalName);
+                }
+            }
+        }
+        extensionTypeFeedItemCounts[placeholderTypeId] += 1;
+    }
+
+    // Create a spreadsheet tab for each extension type, and output its data
+    Logger.log("Outputting data to spreadsheet");
+    for (var placeholderTypeId in jsonAttributeValuesByExtensionType) {
+        var placeholderTable = extensionPlaceholderTypeIdToFieldIdsObjectMapping[placeholderTypeId];
+        var orderedAttributeIds = Object.keys(placeholderTable).sort();
+        orderedAttributeIds.pop();
+
+        var newSheet = spreadsheet.insertSheet(placeholderTable.name);
+        var headerRowValues = Object.keys(placeholderTable).sort().map(
+            function (attributeId) { return placeholderTable[attributeId] }
+        ).slice(0, -1);
+        headerRowValues.push("Feed Item Count", "Reasons for Disapproval");
+        newSheet.appendRow(headerRowValues).getRange(1, 1, 1, headerRowValues.length).setWrap(true);
+
+        var jsonAttributeValuesArray = jsonAttributeValuesByExtensionType[placeholderTypeId];
+        disapprovalShortNamesToOutput = disapprovalShortNamesByExtensionType[placeholderTypeId];
+        var sheetOutput = [];
+        for (var i = 0, n = jsonAttributeValuesArray.length; i < n; i++) {
+            var extensionJsonAttributeValues = jsonAttributeValuesArray[i];
+            var disapprovalShortNames = disapprovalShortNamesToOutput[i];
+            if (Object.keys(duplicateAttributeValuesCounts).indexOf(extensionJsonAttributeValues) !== -1) {
+                var feedItemCount = duplicateAttributeValuesCounts[extensionJsonAttributeValues];
+            } else {
+                var feedItemCount = 1;
+            }
+            var attributeValues = JSON.parse(extensionJsonAttributeValues);
+            var rowValues = orderedAttributeIds.map(
+                function (attributeId) {
+                    if (attributeValues.hasOwnProperty(attributeId)) {
+                        return attributeValues[attributeId].toString();
+                    } else {
+                        return '';
+                    }
+                }
+            );
+            rowValues.push(feedItemCount, disapprovalShortNames.join(','));
+            sheetOutput.push(rowValues);
+        }
+
+        sheetOutput.sort(function (rowValuesA, rowValuesB) {
```

Comment:
What does this sort on/ what does this do? Bit confused!

Reply:
It sorts the extension rows by Feed Item Count, before outputting to the sheet. That is the penultimate value in each row (added in the line above `rowValues.push(feedItemCount, disapprovalShortNames.join(','));`).
I've added a comment.